CriticalMass/Ride.swift

```diff
@@ -16,3 +17,32 @@ struct Ride: Hashable, Codable {
     let estimatedDistance: Double?
     let estimatedDuration: Double?
 }
+
+extension Ride {
+    var coordinate: CLLocationCoordinate2D {
+        CLLocationCoordinate2D(latitude: latitude, longitude: longitude)
+    }
+
+    var titleAndTime: String {
+        let titleWithoutDate = title.removedDatePattern()
+        return "\(titleWithoutDate)\n\(dateTime.humanReadableDate()) - \(dateTime.humanReadableTime())"
+    }
+}
+
+private extension String {
+    func removedDatePattern() -> String {
+        // assuming the date is formatted like dd.MM.yyyy. Also removes the whitespace before the date
+        let pattern = " \\d{1,2}.\\d{1,2}.\\d{4}$"
+        return removedRegexMatches(pattern: pattern)
+    }
+
+    private func removedRegexMatches(pattern: String, replaceWith template: String = "") -> String {
+        do {
+            let regex = try NSRegularExpression(pattern: pattern, options: NSRegularExpression.Options.caseInsensitive)
+            let range = NSRange(location: 0, length: count)
+            return regex.stringByReplacingMatches(in: self, options: [], range: range, withTemplate: template)
+        } catch {
+            return self
+        }
+    }
+}
```

Comment:
would be nice to have some test cases that cover that. Might also be helpful to get a better understanding on what's going on here